python/ray/_private/worker.py

```diff
@@ -867,19 +904,35 @@ def raise_errors(self, serialized_objects, object_refs):
         for e in out:
             _unhandled_error_handler(e)
 
-    def deserialize_objects(self, serialized_objects, object_refs):
+    def deserialize_objects(
+        self, serialized_objects, object_refs, tensor_transport_hint=None
+    ):
         gpu_objects: Dict[str, List["torch.Tensor"]] = {}
         for obj_ref, (_, _, tensor_transport) in zip(object_refs, serialized_objects):
             # If using a non-object store transport, then tensors will be sent
             # out-of-band. Get them before deserializing the object store data.
-            if (
-                tensor_transport is None
-                or tensor_transport == TensorTransportEnum.OBJECT_STORE
-            ):
-                continue
+            tensor_transport_hint_val = (
```

Comment:
Hmm the logic here is getting a bit confusing between tensor_transport_hint, tensor_transport, and obj_ref :D Is there a way to clean it up / Can you add a comment here explaining which tensor transport will actually get used?